CMake/FindMKL.cmake

```diff
@@ -2,42 +2,106 @@
 # This needs a lot of work to make it robust
 INCLUDE( CheckCXXSourceCompiles )
 
+if ( NOT CMAKE_CXX_COMPILER_ID MATCHES "Intel" )
+  # if not intel and MKL_ROOT not set
+  if (NOT MKL_ROOT)
+    find_path(MKL_ROOT "mkl.h"
+      HINTS ${MKLROOT} ${MKL_HOME} $ENV{MKLROOT} $ENV{MKL_ROOT} $ENV{MKL_HOME}
+      PATH_SUFFIXES include)
+    string(REPLACE "/include" "" MKL_ROOT ${MKL_ROOT})
+  endif (NOT MKL_ROOT)
```

Comment:
Open an ELSE and move the later MKL_INCLUDE_DIRECTORIES here. I don't think we need to do it twice.

Reply:
My prefered use case is that MKL_ROOT is simply set and we don't depend on any environment variables.  However this section will support some mkl module setups I've seen.  I am not interested in making the main execution path more convoluted for to support this.

Reply:
I see. Then leave it this way. Not here but in the later lines, could you add a bit more comment saying "detecting library" "detecting header files" "setting variables"? This will help readability.